### PR TITLE
Fixed dependency on SecurityContextInterface

### PR DESCRIPTION
**Pull request by Xymanek**
I am targeting this branch, because this is aimed at sf 3+.

## Changelog

```markdown
### Fixed
- Fixed missing security.context service error
```

### CODE REVIEW

**Comment by kunicmarko20** [comment]
Please write a changelog directed at end users. Keep in mind that they probably do not care what file or class you changed, but what features you provided / what bugs you fixed. Try to answer the question "What benefit do I get from upgrading?"
More on http://keepachangelog.com

File: src/OrderBundle/Block/RecentOrdersBlockService.php [hunk 1]
```diff
@@ -73,7 +76,7 @@ public function execute(BlockContextInterface $blockContext, Response $response
         $criteria = [];
 
         if ('admin' !== $blockContext->getSetting('mode')) {
-            $orders = $this->orderManager->findForUser($this->securityContext->getToken()->getUser(), ['createdAt' => 'DESC'], $blockContext->getSetting('number'));
+            $orders = $this->orderManager->findForUser($this->tokenStorage->getToken()->getUser(), ['createdAt' => 'DESC'], $blockContext->getSetting('number'));
```
Comment (kunicmarko20) [inline src/OrderBundle/Block/RecentOrdersBlockService.php]: Could you break this line, we follow limit of 120 characters

Comment (Xymanek) [reply src/OrderBundle/Block/RecentOrdersBlockService.php]: I though about it, but running `php-cs-fixer` didn't change that....

Will fix now

File: src/OrderBundle/Resources/config/block.xml [hunk 2]
```diff
@@ -7,8 +7,8 @@
             <argument type="service" id="templating"/>
             <argument type="service" id="sonata.order.manager"/>
             <argument type="service" id="sonata.customer.manager"/>
-            <argument type="service" id="security.context"/>
-            <argument type="service" id="sonata.admin.pool" on-invalid="ignore"/>
+            <argument type="service" id="security.token_storage"/>
+            <argument type="service" id="sonata.admin.pool" on-invalid="null"/>
```
Comment (jordisala1991) [inline src/OrderBundle/Resources/config/block.xml]: why the on-invalid?

Comment (Xymanek) [reply src/OrderBundle/Resources/config/block.xml]: No idea, it was there before

**Comment by jordisala1991** [comment]
Can you review this one @greg0ire ? Do we need an upgrade note here?

**Comment by greg0ire** [comment]
> Do we need an upgrade note here?

Yes. If it is major, then an action should be taken by the user after upgrading. This action should be described.

@Xymanek, please avoid "fix" in commit messages. Instead, describe precisely what you did and why you did it. More info in the saved reply below:

Please format your commit message according to our rules : 

- The commit message subject must be less than 50 characters and tell us *what* you did.
- The commit message body should tell us *why* you did it. It is optional but highly recommended.

Bad example :

```
Fixed bug #989 by removing call to defraculate()
```

Good example:

```
Remove call to defraculate()

Calling this function caused a bug because it interferes 
with calls to getSchmeckles().
Fixes #989
```

More details in CONTRIBUTING.md

**Comment by Xymanek** [comment]
> If it is major, then an action should be taken by the user after upgrading

What action? I just used new sf security API instead of old one in an internal class - this should be transparent to users (developers)

**Comment by greg0ire** [comment]
If it's transparent it is not major. But the signature changes, so it is a BC-break, and the action to take is to stop calling that class with the wrong argument if you do, and to change any class that inherits from this one if there are.

**Comment by Xymanek** [comment]
> stop calling that class with the wrong argument if you do, and to change any class that inherits from this one if there are.

You are right, I didn't considered these cases... In that case does it also make sense to remove ` = null` from `$adminPool`?

**Comment by greg0ire** [comment]
I think it does

File: src/OrderBundle/Block/RecentOrdersBlockService.php [hunk 3]
```diff
@@ -43,23 +43,26 @@ class RecentOrdersBlockService extends BaseBlockService
...
-    {
+    protected $adminPool;
+
+    public function __construct(
+        string $name,
+        EngineInterface $templating,
+        OrderManagerInterface $orderManager,
+        CustomerManagerInterface $customerManager,
+        TokenStorageInterface $tokenStorage,
+        ?Pool $adminPool
```
Comment (jordisala1991) [inline src/OrderBundle/Block/RecentOrdersBlockService.php]: where is this used?

Comment (jordisala1991) [reply src/OrderBundle/Block/RecentOrdersBlockService.php]: oh I see it now

Comment (kunicmarko20) [reply src/OrderBundle/Block/RecentOrdersBlockService.php]:
```
return $this->renderPrivateResponse($blockContext->getTemplate(), [
             'context' => $blockContext,
             'settings' => $blockContext->getSettings(),
             'block' => $blockContext->getBlock(),
             'orders' => $orders,
             'admin_pool' => $this->adminPool,
         ], $response);
     }
```

File: src/OrderBundle/Resources/config/block.xml (hunk 2 above)
Comment (jordisala1991) [reply src/OrderBundle/Resources/config/block.xml]: If you remove the null on the construct you need to remove the on-invalid here

**Comment by kunicmarko20** [comment]
The constructor is nullable (?Pool), it just isn't null by default

**Comment by jordisala1991** [comment]
Omg so many errors by me on this one.

Anyway I think that could be removed, it was added here: https://github.com/sonata-project/ecommerce/commit/cc8c1bb8831adb46bc6a483fe6f442d91da2235c

But the admin-bundle is a dependency on this bundle, so I guess the admin pool will be always be there... wdyt?

**Comment by kunicmarko20** [comment]
Makes sense to remove it if we will always have the pool.

**Comment by Xymanek** [comment]
Well it does check if pool is present or not in the template, so dunno...

https://github.com/Xymanek/ecommerce/blob/ef5af32faf5f560e437d3ee3d416f163395588e8/src/OrderBundle/Resources/views/Block/recent_orders.html.twig#L31

https://github.com/Xymanek/ecommerce/blob/ef5af32faf5f560e437d3ee3d416f163395588e8/src/OrderBundle/Resources/views/Block/recent_orders.html.twig#L53

**Comment by jordisala1991** [comment]
Let's not remove the `on-invalid` then for now, it can be done later :P

**Comment by jordisala1991** [comment]
Thank you @Xymanek

**Pull request merged**